src/mds/agg_mds/adapters.py

```diff
@@ -156,12 +190,28 @@ def mapFields(item: dict, mappings: dict, global_filters=None) -> dict:
             elif isinstance(value, str) and "path:" in value:
                 # process as json path
                 expression = value.split("path:")[1]
-                field_value = get_json_path_value(expression, item)
+
+                hasDefaultValue = False
+                default_value = None
+                if key in schema:
+                    d = schema[key].default
+                    if d is not None:
+                        hasDefaultValue = True
+                        default_value = d
+
+                field_value = get_json_path_value(
+                    expression, item, hasDefaultValue, default_value
+                )
             else:
                 field_value = value
 
             for f in global_filters:
                 field_value = FieldFilters.execute(f, field_value)
+            if key in schema:
+                field_value = schema[key].normalize_value(field_value)
+            # set to default if conversion failed and a default value is available
+            if field_value is None and hasDefaultValue:
```

Comment:
can we handle this more elegantly?

AggMDS sync is failing in HEAL preprod because of error `RequestError(400, 'mapper_parsing_exception', 'object mapping for [advSearchFilters] tried to parse field [advSearchFilters] as object, but found a concrete value')`

I think it is because some metadata doesn't have this `advSearchFilters` field
And we should be adding the field with the define default value to the ES doc in this case, so the job won't crash.

Reply:
I added a log warning. However, in this case, the field was not in the schema and had no default value in the config file. This issue should be addressed in the config